sourcegraph/sourcegraph
# bext: refactor integration tests

Closes https://github.com/sourcegraph/sourcegraph/issues/33484

1. Changes browser extension integration tests snapshots compression mechanism. We used `git lfs` to store snapshots before, but used too much bandwidth to fetch them. This PR suggests compressing the snapshots after recording and adding compressed files to repo index. Snapshots are decompressed before and removed after running integration tests.
2. Enables browser extension integration tests in the pipeline as we no longer fetch snapshots with `git-lfs`.
3. Removes browser extension after install page integration test as neither Polly nor Percy support non _http(s)_ protocols.
4. Refactors GitHub search results pages integration tests to omit 429 HTTP errors causing recordings to fail. There's still no guarantee we won't have this error while recording all integration tests at once in the future.

## Test plan
- run `sg test bext-build`
- run `yarn workspace @sourcegraph/browser record-integration` and ensure that 
  - it passes 
  - there are no test recording files (`.har`) in git index
  - compressed recordings (`.har.gz`) are tracked by git  
- run `sg test bext-integration` and ensure that
  - tests pass
  - there are no test recording files (`.har`) in git index




## App preview:
- [Link](https://sg-web-taras-yemets-refactor-bext-tests.onrender.com)
